### PR TITLE
Fix the broken logic of moving elements forward in a sequence (Blaze)

### PR DESCRIPTION
**Pull request by Slava**
We had a bug in `observe-sequence`, a translation layer between the array diff and the DOM operations.

Can someone from the Blaze team review it for me? @avital @stubailo @dgreensp

### CODE REVIEW

**Comment by Slava** [comment]
See #2845 for the context and a reproduction.

File: packages/observe-sequence/observe_sequence.js [hunk 1]
```diff
@@ -195,13 +195,28 @@ var diffArray = function (lastSeqArray, seqArray, callbacks) {
...
+      // The original issued callback is "1" before "4".
+      // The position of "1" is 1, the position of "4" is 4.
+      // The generated move is (1) -> (3)
+      if (position > prevPosition) {
+        position--;
+      }
 
       _.each(posCur, function (pos, id) {
-        if (pos >= prevPosition && pos <= position)
+        if (pos > prevPosition && pos < position)
```
Comment (stubailo) [inline packages/observe-sequence/observe_sequence.js]: What is `pos` and `position`? we should probably rename the `position` variable to something more descriptive. is it `newPosition`?

**Comment by Slava** [comment]
Thanks @stubailo, merging to devel.

**Comment by dgreensp** [comment]
Nice tests!

**Pull request merged**